pkg/controller/datadogagent/clusteragent.go

```diff
@@ -451,17 +451,10 @@ func getEnvVarsForClusterAgent(dda *datadoghqv1alpha1.DatadogAgent) []corev1.Env
 	}
 
 	if needClusterAgentSecret(dda) {
-		if spec.Credentials.APIKeyExistingSecret != "" {
```

Comment:
we should keep supporting this!

Reply:
could you add a test case when users specify an existing secret object?

Reply:
The feature of user provided secret for the API key is still there because this test was [already present in `GetAPIKeySecretName(…)`](https://github.com/DataDog/datadog-operator/blob/lenaic/fix_cred_in_secret/pkg/controller/utils/shared_utils.go) that is used by the `getAPIKeyFromSecret(…)` function that is now systematically used here.
The only difference is that now, the API key is always coming from a secret, but it can still either be the one provided by the operator or the user-supplied one.

Reply:
great thanks, is it covered in the unit tests? if it's not the case could you please add a test case for it ?

Reply:
It wasn’t. I added one to test this use case.

Reply:
thanks a lot! 💯